src/util/filename.cpp

```diff
@@ -0,0 +1,30 @@
+#include "util/filename.h"
+
+#include <QStringList>
+
+namespace mixxx {
+
+namespace filename {
+
+QString sanitize(const QString& unsanitizedName) {
+    QString sanitizedName = unsanitizedName;
+    sanitizedName.replace("/", "-");
```

Comment:
Why is the forward slash replace but the other chars are removed?

How about replacing a regex like `([<>:\\/"'\|\?\*]+)` with a single `-`? I don't know if QString allow it, but all ASCII control characters (0-31) are forbidden on Windows, too.

On windows, filenames may also not end with a space or a dot.
Also, the following filenames are forbidden (both standalone and with arbitrary extensions): 
```
CON, PRN, AUX, NUL 
COM1, COM2, COM3, COM4, COM5, COM6, COM7, COM8, COM9
LPT1, LPT2, LPT3, LPT4, LPT5, LPT6, LPT7, LPT8, LPT9
```

See: https://stackoverflow.com/a/31976060

Reply:
![XKCD comic](https://imgs.xkcd.com/comics/regular_expressions.png)

Reply:
Done in 90426fccec5961777b5bc62f4348957256e1151c. I used a QStringList instead of a regex for the individual characters because the regex got quite confusing with all the escaping (even when using a C++ raw string).